src/articles/comprehensive-journey-to-elm.md

```diff
@@ -0,0 +1,172 @@
+update: Msg -> Model -> Model
+update msg model =
+  -- A case is like a switch in JS, except you have to handle each possible case
+  case msg of
+    Increment ->
+      model + 1
+
+    Decrement ->
+      model - 1
+
+-- Our view is a function of our model
+-- No HTML, you just use Elm to create the view (a bit like using JSX)
+view model =
+  div []
+    -- You can bind messages that will be given to your update function on some events
+    [ button [ onClick Decrement ] [ text "-" ]
+    , div [] [ text (toString model) ]
+    , button [ onClick Increment ] [ text "+" ]
+    ]
+```
+
+The [Elm Architecture](https://guide.elm-lang.org/architecture/) is the way you structure your Elm applications, with three distinctive parts:
+
+ - a model, which represents the state of our application;
+ - a view function, that generate the view from the model;
+ - an update function to perform model changes.
+
+When you want to change your model, you send an *action* that will be given to your `update` function. The update function is the only place your model can be changed. The view is then re-rendered if needed. As each Elm application is based on that model and as the formatting is standardized by [elm-format](https://github.com/avh4/elm-format), it is really easy to jump into another dev's codebase.
+
+It seems important to consider the ecosystem before adopting a language. For Elm, it was a really good surprise. The tooling is really good! You can easily bundle your code with [elm-make](https://github.com/elm-lang/elm-make) and start your dev ecosystem with [elm-reactor](https://github.com/elm-lang/elm-reactor). A [REPL](https://github.com/elm-lang/elm-repl) is also available. But to really industrialize your project, you can use some solutions like [create-elm-app](https://github.com/halfzebra/create-elm-app), inspired by *create-react-app*. The community is active and the [slack](http://elmlang.herokuapp.com/) is crowded with people willing to help.
+
+Testing is conveniently easy, thanks to [elm-test](https://github.com/elm-community/elm-test). For example, this is all you need to test the counter code above:
+
+```elm
+module Tests exposing (..)
+
+import Test exposing (..)
+import Expect
+import App exposing (update, Msg(Increment, Decrement))
+
+
+all : Test
+all =
+    describe "Test suite"
+        [ test "Increment" <|
+            \() ->
+                Expect.equal (update Increment 7) 8
+        , test "Decrement" <|
+            \() ->
+                Expect.equal (update Decrement -3) -4
+        ]
+```
+
+There are already [many packages available](http://package.elm-lang.org/). I know, we are far from the 350,000+ JavaScript modules on npm, but this is another advantage of Elm: the language is fully interoperable with JavaScript! [Using ports](https://guide.elm-lang.org/interop/javascript.html#ports), you can communicate between the two languages and therefore reuse any existing JS module.
```

Comment:
we are, je pense pas que ça se dise en anglais. Enfin je me prenais des remarques quand j'écrivais ça à l'école :p